### PR TITLE
esp_tinyusb: set cpu core affinity of tinyusb default task

### PR DESCRIPTION
**Pull request by esp-saurabhbansal**
1. This change is to set the CPU core affinity to Tinyusb Task.
If the task is not pinned we often see lost packets if the system is put under stress.

2. cdc console feature is enabled only when VFS is enabled.
This is done by compiling the console feature related files only when VFS option is enabled.
If VFS is not enabled, compile error will be thrown. So, it will avoid runtime failure.
Closes https://github.com/espressif/idf-extra-components/issues/164

### CODE REVIEW

**Comment by esp-saurabhbansal** [comment]
@tore-espressif @roma-jam @Dazza0 
Please check this change.

**Comment by tore-espressif** [comment]
@esp-saurabhbansal Thank you for the PR, I was thinking about making all the task related configs (stack size, affinity, priority) run-time configurable by extending `tinyusb_config_t`. With backward compatibility.

What do you think?

File: usb/esp_tinyusb/tusb_tasks.c [hunk 1]
```diff
@@ -12,6 +12,14 @@
 #include "tinyusb.h"
 #include "tusb_tasks.h"
 
+#if !CONFIG_TINYUSB_NO_DEFAULT_TASK
```
Comment (roma-jam) [inline usb/esp_tinyusb/tusb_tasks.c:15]: Nitpick: it is sometimes hard to understand what is the goal when there are two or more denial in logic. 
I mean, here we have (NOT)*_NO_DEFAULT. 
Probably, there is a chance to change it to #if CONFIG_TINYUSB_DEFAULT_TASK.

Comment (esp-saurabhbansal) [reply usb/esp_tinyusb/tusb_tasks.c]: Thanks @roma-jam for the comment.
All the CONFIG options for background tinyusb task are removed now. Instead, runtime parameters are used.

**Comment by esp-saurabhbansal** [comment]
> @esp-saurabhbansal Also, could you please update this line https://github.com/espressif/idf-extra-components/blob/master/usb/esp_tinyusb/CMakeLists.txt#L49 like this:
> 
> ```
> # Determine whether tinyusb is fetched from component registry or from local path
> idf_build_get_property(build_components BUILD_COMPONENTS)
> if(tinyusb IN_LIST build_components)
>     set(tinyusb_name tinyusb) # Local component
> else()
>     set(tinyusb_name espressif__tinyusb) # Managed component
> endif()
> 
> # Pass tusb_config.h from this component to TinyUSB
> idf_component_get_property(tusb_lib ${tinyusb_name} COMPONENT_LIB)
> ```

This is done.

**Comment by esp-saurabhbansal** [comment]
> @esp-saurabhbansal Thank you for the PR, I was thinking about making all the task related configs (stack size, affinity, priority) run-time configurable by extending `tinyusb_config_t`. With backward compatibility.
> 
> What do you think?

Done

File: usb/esp_tinyusb/include/tinyusb.h [hunk 2]
```diff
@@ -34,6 +34,10 @@ typedef struct {
     const uint8_t *configuration_descriptor;   /*!< Pointer to a configuration descriptor. If set to NULL, TinyUSB device will use a default configuration descriptor whose values are set in Kconfig */
     bool self_powered;                         /*!< This is a self-powered USB device. USB VBUS must be monitored. */
     int vbus_monitor_io;                       /*!< GPIO for VBUS monitoring. Ignored if not self_powered. */
+    bool no_backround_task;                    /*!< When set to false, background task handling tinyusb events is created. */
```
Comment (tore-espressif) [inline usb/esp_tinyusb/include/tinyusb.h:37]: type
```suggestion
    bool no_background_task;                    /*!< When set to false, background task handling tinyusb events is created. */
```

File: usb/esp_tinyusb/tinyusb.c [hunk 3]
```diff
@@ -92,9 +95,24 @@ esp_err_t tinyusb_driver_install(const tinyusb_config_t *config)
...
+        if (config->task_priority != 0) {
+            task_priority = config->task_priority;
+        } else {
+            task_priority = 5; // Backward compatibility with esp_tinyusb v1.0.0. Do NOT remove!
+        }
+        if (config->core_id >= 0 && (config->core_id <= configNUM_CORES - 1)) {
+            core_id = config->core_id;
+        } else {
+            core_id = tskNO_AFFINITY; // Backward compatibility with esp_tinyusb v1.0.0. Do NOT remove!
+        }
```
Comment (tore-espressif) [inline usb/esp_tinyusb/tinyusb.c:113]: Actually, we want to pin the task to core 0 by default. There are issues when the task is not pinned, so before we fix it, the task should be pinned be default.

File: usb/esp_tinyusb/include/tinyusb.h [hunk 4]
```diff
@@ -34,6 +34,10 @@ typedef struct {
...
+    uint32_t task_priority;                    /*!< Task priority of created background tinyusb task */
+    UBaseType_t stack_size;                    /*!< Stack size of crated background tinyusb task */
+    BaseType_t core_id;                        /*!< Select core on which background tinyusb task will run or tskNO_AFFINITY  */
```
Comment (tore-espressif) [inline usb/esp_tinyusb/include/tinyusb.h:40]: In most of our components we use type `size_t`. This way, we don't have to include freertos headers in this file

File: usb/esp_tinyusb/tinyusb.c (hunk 3 above)
Comment (esp-saurabhbansal) [reply usb/esp_tinyusb/tinyusb.c:113]: By default, config->core_id is 0 and task will be pinned to core 0. So, the implementation is correct. Only line to change is:
`core_id = tskNO_AFFINITY; // Backward compatibility with esp_tinyusb v1.0.0. Do NOT remove!`
->
`core_id = tskNO_AFFINITY;`

is it OK ?

File: usb/esp_tinyusb/include/tinyusb.h (hunk 2 above)
Comment (esp-saurabhbansal) [reply usb/esp_tinyusb/include/tinyusb.h]: done

File: usb/esp_tinyusb/include/tinyusb.h (hunk 4 above)
Comment (esp-saurabhbansal) [reply usb/esp_tinyusb/include/tinyusb.h]: I have changed the type of `task_priority` and `stack_size` to `size_t`
For core_id, I have kept it `BaseType_t `

**Comment by tore-espressif** [comment]
@esp-saurabhbansal Thank you for the changes. But we will delay this breaking change until next major refactoring

**Pull request closed without merging**